packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/__tests__/quanticFeedbackModalQna.test.js

```diff
@@ -0,0 +1,299 @@
+      selectors.questionTopicRadio
+    );
+    expect(questionTopicRadio).not.toBeNull();
+
+    const questionHallucinationRadio = element.shadowRoot.querySelector(
+      selectors.questionHallucinationRadio
+    );
+    expect(questionHallucinationRadio).not.toBeNull();
+
+    const questionDocumentedRadio = element.shadowRoot.querySelector(
+      selectors.questionDocumentedRadio
+    );
+    expect(questionDocumentedRadio).not.toBeNull();
+
+    const questionReadableRadio = element.shadowRoot.querySelector(
+      selectors.questionReadableRadio
+    );
+    expect(questionReadableRadio).not.toBeNull();
+
+    const documentUrlInput = element.shadowRoot.querySelector(
+      selectors.documentUrlInput
+    );
+    expect(documentUrlInput).not.toBeNull();
+
+    const additionalDetailsInput = element.shadowRoot.querySelector(
+      selectors.additionalDetailsTextarea
+    );
+    expect(additionalDetailsInput).not.toBeNull();
+
+    const skipButton = element.shadowRoot.querySelector(selectors.skipButton);
+    expect(skipButton).not.toBeNull();
+
+    const submitButton = element.shadowRoot.querySelector(
+      selectors.submitButton
+    );
+    expect(submitButton).not.toBeNull();
+  });
+
+  describe('when submitting the feedback', () => {
+    describe('with missing required answers', () => {
+      it('should not submit the feedback and not show the success screen', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        // Mock the checkValidity method to return false since the questions are not answered.
+        const checkValidityMock = jest.fn(() => false);
+        const answerEvaluationQuestions = element.shadowRoot.querySelectorAll(
+          selectors.answerEvaluationQuestions
+        );
+        answerEvaluationQuestions.forEach((question) => {
+          question.checkValidity = checkValidityMock;
+        });
+
+        const submitButton = element.shadowRoot.querySelector(
+          selectors.submitButton
+        );
+        submitButton.click();
+        await flushPromises();
+
+        const successMesage = element.shadowRoot.querySelector(
+          selectors.successMessage
+        );
+        expect(successMesage).toBeNull();
+        expect(checkValidityMock).toHaveBeenCalledTimes(4);
+      });
+    });
+
+    describe('with all the required questions answered', () => {
+      it('should display the success message and execute the handleSubmit function', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        lightningElementValueChange(
+          element,
+          selectors.questionTopicRadio,
+          'yes'
+        );
+        lightningElementValueChange(
+          element,
+          selectors.questionDocumentedRadio,
+          'yes'
+        );
+        lightningElementValueChange(
+          element,
+          selectors.questionHallucinationRadio,
+          'yes'
```

Comment:
I wouldn't put all the expected answers to `yes` to also check that our component can return other values as well and not always `yes`  whatever what option you select.

Reply:
It doesn't matter which value I put, I'm the one setting the value.

So even if I put 'abracadabra', in the end I'm the one that says `value='abracadabra'` and then executing `get value` so essentially I can only test that the code gets the value I set. Changing the string doesn't add anything to the test.

Keep in mind, in reality I am not clicking, nor selecting an option, I am setting the `lightning-radio-group` stub's `value` attribute to any string I want. And when the code of `quanticFeedbackModalQna` checks for the value, it simply returns the value I've set on the stub.

Reply:
That's where we feel the value of E2E tests to come and complement the unit tests.